modules/uac_registrant/registrant.c

```diff
@@ -518,6 +520,16 @@ int run_reg_tm_cback(void *e_data, void *data, void *r_data)
 				contact = contact->next;
 			}
 		}
+		// if there is no expires in contact, try parse expires from header
+		if (exp == 0 && msg->expires) {
+			if (parse_expires(msg->expires) < 0) {
+				LM_ERR("failed to parse Expires header\n");
+			}
+			header_exp = (exp_body_t*)msg->expires->parsed;
+			if (header_exp && header_exp->valid) {
```

Comment:
LGTM, but you could improve here by avoiding the double testing and have all the ops with `header_exp` on the `else` branch of `if (parse_expires(msg->expires) < 0) {`